tools/ReleaseTool/ReleaseCommand.cs

```diff
@@ -0,0 +1,216 @@
+﻿using CommandLine;
+using System;
+using System.IO;
+using System.Text;
+using Octokit;
+
+namespace ReleaseTool
+{
+    /// <summary>
+    /// Runs the commands required for releasing a candidate.
+    /// * Does a final bump of the gdk.pinned (if needed).
+    /// * Merges the candidate branch into develop.
+    /// * Pushes develop to origin/develop and origin/master.
+    /// * Creates a GitHub release draft.
+    /// </summary>
+    internal class ReleaseCommand
+    {
+        private const string UpdateGdkCommitMessageTemplate = "Update to release commit for Gdk {0}.";
+        private const string MergeCommitMessageTempalte = "Merge release {0}.";
+
+        private const string PackageContentType = "application/zip";
+        private const string ChangeLogFilename = "CHANGELOG.md";
+
+        [Verb("release", HelpText = "Merge a release branch and create a github release draft.")]
+        public class Options: GitClient.IGitOptions, GitHubClient.IGitHubOptions
+        {
+            [Value(0, MetaName = "version", HelpText = "The version that is being released.")]
+            public string Version { get; set; }
+
+            [Option('g', "update-gdk", HelpText = "The git hash of the version of the gdk to upgrade to. (Only if this is a project).")]
+            public string GdkVersion { get; set; }
+
+            public string GitRemote { get; set; }
+
+            public string DevBranch { get; set; }
+
+            public string MasterBranch { get; set; }
+
+            public string GitHubTokenFile { get; set; }
+        }
+
+        private readonly Options options;
+
+        public ReleaseCommand(Options options)
+        {
+            this.options = options;
+        }
+
+        public int Run()
+        {
+            var gitClient = new GitClient(options);
+            var gitHubClient = new GitHubClient(options);
+
+            try
+            {
+                if (gitClient.HasStagedOrModifiedFiles())
+                {
+                    throw new InvalidOperationException("The repository currently has files staged, please stash, reset or commit them.");
+                }
+
+                gitHubClient.LoadCredentials();
+                var gitHubRepo = gitHubClient.GetRepositoryFromRemote(gitClient.GetRemoteUrl());
+
+                // Checkout "origin/feature/release-X"
+                var branchName = BranchName();
+                gitClient.Fetch();
+                gitClient.CheckoutRemoteBranch(branchName);
+
+                // Make Changes if necessary
+                UpdateGdkVersion(gitClient);
+
+                // Merge into develop
+                Console.WriteLine($"Merging {branchName} into {options.DevBranch}...");
+                var branchCommit = gitClient.GetHeadCommit();
+                gitClient.CheckoutRemoteBranch(options.DevBranch);
+                gitClient.Merge(branchCommit, string.Format(MergeCommitMessageTempalte, options.Version));
```

Comment:
Is this a squash merge?

Reply:
Also this bypasses our CI gates? 

Can this merge the open PR into develop instead?

Reply:
So the api for merging an open PR is a very finicky. (We would have to search through all PRs and look for the correct one).

There are a few options:
* We could leave as is and (unfortunately) this will bypass our merge gates -- but we could put build dependencies on BuildKite.
* We could make the PR web address an argument.
* We could search through all open PRs (potentially difficult).

Which do you prefer?

Willn also change this to a Squash Merge.

Reply:
Could you give a PR id as an argument? Rather than the full url?